app/javascript/gobierto_data/webapp/components/sets/data/SQLEditorResults.vue

```diff
@@ -127,54 +126,60 @@ export default {
       labelVisualize: I18n.t('gobierto_data.projects.visualize') || "",
       labelResetViz: I18n.t('gobierto_data.projects.resetViz') || "",
       labelModifiedVizualition: I18n.t("gobierto_data.projects.modifiedVisualization") || "",
+      labelSavedVisualization: I18n.t("gobierto_data.projects.savedVisualization") || "",
       showVisualization: false,
-      isVisualizationModified: false,
       showResetViz: false,
       showVisualize: true,
       removeLabelBtn: false,
       perspectiveChanged: false,
-      typeChart: 'hypergrid'
+      typeChart: 'hypergrid',
     };
   },
+  watch: {
+    vizInputFocus(newValue) {
+      if (newValue) {
+        this.$nextTick(() => this.$refs.savingDialogViz.inputFocus())
+      }
+    }
+  },
   methods: {
     onSaveEventHandler(opts) {
       // get children configuration
       const config = this.$refs.viewer.getConfig()
-
       this.$root.$emit("storeCurrentVisualization", config, opts);
-      this.removeLabelBtn = true
-      this.isVisualizationModified = false
+    },
+    updateVizName(value) {
+      const {
+        name: vizName
+      } = value;
+      this.labelValue = vizName
+      this.$root.$emit("eventIsVizModified", true);
     },
     resetViz() {
       const hidePerspective = "none"
 
       this.showVisualize = true
       this.perspectiveChanged = false
-      this.removeLabelBtn = false
       this.showResetViz = false
       this.typeChart = 'hypergrid'
-      this.isVisualizationModified = false
 
       this.$refs.viewer.enableDisabledPerspective(hidePerspective);
       this.$refs.viewer.setColumns();
+      this.$root.$emit('enableSavedVizButton', true)
+      this.$root.$emit('disabledSavedVizString')
```

Comment:
That's why I insist on this kind of events: you're resetting a visualization, why do you enable a button and disabled a string? why a function called `resetViz` has to know this sort of things?

The workflow should be the following: you emit "reset the viz", whoever is listening such event KNOWS that: the saving button must be enabled, AND the string disabled.